pkg/logger/logger.go

```diff
@@ -568,3 +568,28 @@ func GetLevel() Level {
 func HasDebugLevel() bool {
 	return pkgLogger.cfg.Level == DebugLevel
 }
+
+// ErrorFuncName displays a given error prefixed by the current (caller) fn name
+// and a given string in a given format (just like fmt.Sprintf)
+func NewErrorf(format string, a ...interface{}) error {
+	sentence := fmt.Sprintf(format, a...)
+
+	pc, _, _, _ := runtime.Caller(1)
+	funcName := runtime.FuncForPC(pc).Name()
+	index := strings.LastIndex(funcName, "/") + 1
+
+	return fmt.Errorf("%v: %v", funcName[index:], sentence)
+}
+
+// ErrorFunc displays a given error prefixed by the current (caller) fn name
+func ErrorFunc(e error) error {
+	if e != nil {
+		pc, _, _, _ := runtime.Caller(1)
+		funcName := runtime.FuncForPC(pc).Name()
+		index := strings.LastIndex(funcName, "/") + 1
+
+		return fmt.Errorf("%v: %v", funcName[index:], e)
+	}
+
+	return nil
+}
```

Comment:
This is a great improvement for debugging without doubts. It's awesome, thanks @rafaeldtinoco.

I'm ok with this PR and already approving it, but I still have some concerns about the naming and where these helpers lie. Reading the returns with the enriched errors, I got a bit confused about the meaning and origin. It looks like we are returning a logger function to be managed/called from the caller.

What about to move them to `pkg/{err,er,e,te,whatever}/errors.go` and use them like:

`return e.NewErrorf("%s blah", str)`

what could make possible to return it to the previous naming:

`return e.Errorf("%s blah", str)`

I think that could make the purpose clear.

The final suggestion would change `ErrorFunc()` to something like `WrapError(err)`, `EnrichError(err)` or `FormatError(err)`.

Or as it would always be used directly from the error pkg, something as `e.Func(err)` would suffix as well.

WDYT?

Reply:
I want to have this released so we can better track issues (much better than what we have today). I'll open an issue to track what you're proposing because I think it is a good idea, but it would take too much time for this release I'm afraid.

Reply:
https://github.com/aquasecurity/tracee/pull/2773#discussion_r1119379506